### DISCUSSION TITLE
Link a file field to a specific folder in the file library

### DISCUSSION THREAD

**Discussion by martijn-dev** [directus/directus#8998] [Feature Requests]
When I create a file field on a collection I can already configure in which folder I would like to place uploaded files (under Interface/Folder). Nice feature!

But when I am in edit mode and I choose "Choose File from Library" on a file field; it opens a side view with all the content in the file library. It would be nice to let this be only the content in the folder that I linked to to this file field (in the configuration).

The way of implementing this is open for discussion. Some discussion points:
- Should you use the same (Interface/Folder) setting for this "Choose File from Library" view or create a new setting for it.
- Should you restrict the field to only take files from this specific folder or allow it to grap content from other folders as well. (Or make this a setting as well)
- Maybe this is something that would fit under the "Conditions" section.

This feature would be extremely useful for big projects with a lot of content, which right now could become a quite big mess.

Let me know what you guys think!

Cheers,
Martijn

**Comment by benhaynes** [comment]
I think we would automatically default to the upload folder when selecting a new item, and then we can add a toggle for force this folder only. The latter might be tricky though, since they still need to be able to traverse subfolders, I assume, but not navigate up towards the root beyond the selected folder. 🤔

**Reply by martijn-dev** [reply]
To me, this would be the ideal outcome for this feature indeed.

In the current 'select file from library' side view there isn't any hierarchy structure at all (just a dump of all files of all folder), so this might need some work first and then the above feature could fit into that.

**Reply by azrikahar** [reply]
@martijn-dev just to get feedback from you in this matter, does #9285 help? It's pretty basic at the moment, but the intention is just to get one foot in the door for this topic since it took minimal additions to make that PR work.

**Reply by martijn-dev** [reply]
@azrikahar Awesome! Yes, this change seems to suit my usecase perfectly. 
As @benhaynes already said there might be some tricky edge cases like:
- What happens with file in a subfolder of the selected folder, would these also be visible? In my opinion that would be logic.
- Should we give a user the freedom to still pick files from another folder, even if a folder is set. (or make this a setting on the field).
- And in adition to the above suggestion, the file select drawer does not have any folder structure. This would be really useful for projects with lots of content, also if you decide to not link a field to a specific folder.

That said, these things are just future thoughts, #9285 is a perfect solution for most of the usecases. And like you said, good to get one foot in the door for this topic! Thanks a lot for this change!

**Reply by azrikahar** [reply]
@martijn-dev thanks for the detailed response!

> What happens with file in a subfolder of the selected folder, would these also be visible? In my opinion that would be logic.

Personally ya this makes sense to me, but I also understand there's still a possibility where some users don't want this to be the case.

> Should we give a user the freedom to still pick files from another folder, even if a folder is set. (or make this a setting on the field).

Originally I did thought of this when I implemented that PR, and I imagined that it "auto-fills" the filter but still let us remove it. However seems like that filter is sort of baked in at the moment, at least without introducing more changes to the codebase 🤔

 > And in adition to the above suggestion, the file select drawer does not have any folder structure. This would be really useful for projects with lots of content, also if you decide to not link a field to a specific folder.
 
 This is definitely useful, and is discussed over at #3464 (the design mockup shared there) as well as on the Discord IIRC.

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
